cordapp-example/clients/src/main/kotlin/com/example/server/MainController.kt

```diff
@@ -0,0 +1,122 @@
+val SERVICE_NAMES = listOf("Notary", "Network Map Service")
+
+
+/**
+ *  A Spring Boot Server API controller for interacting with the node via RPC.
+ */
+
+@RestController
+@RequestMapping("/spring/api/") // The paths for GET and POST requests are relative to this base path.
+class MainController(
+        private val rpc: NodeRPCConnection,
+        @Value("\${$CONTROLLER_NAME}") private val controllerName: String) {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(RestController::class.java)
+    }
+
+    private val myLegalName = rpc.proxy.nodeInfo().legalIdentities.first().name
+    private val proxy = rpc.proxy
+
+    /**
+     * Returns the node's name.
+     */
+    @GetMapping(value = "me", produces = arrayOf(MediaType.APPLICATION_JSON_VALUE))
+    fun whoami() = mapOf("me" to myLegalName)
+
+    /**
+     * Returns all parties registered with the [NetworkMapService]. These names can be used to look up identities
+     * using the [IdentityService].
+     */
+    @GetMapping(value = "peers", produces = arrayOf(MediaType.APPLICATION_JSON_VALUE))
+    fun getPeers(): Map<String, List<CordaX500Name>> {
+        val nodeInfo = proxy.networkMapSnapshot()
+        return mapOf("peers" to nodeInfo
+                .map { it.legalIdentities.first().name }
+                //filter out myself, notary and eventual network map started by driver
+                .filter { it.organisation !in (SERVICE_NAMES + myLegalName.organisation) })
+    }
+
+
+    /**
+     * Displays all IOU states that exist in the node's vault.
+     */
+    @GetMapping(value = "ious", produces = arrayOf(MediaType.APPLICATION_JSON_VALUE))
+    fun getIOUs() : ResponseEntity<List<StateAndRef<IOUState>>> {
+        return ResponseEntity.ok(proxy.vaultQueryBy<IOUState>().states)
+    }
+
+
+    /**
+     * Initiates a flow to agree an IOU between two parties.
+     *
+     * Once the flow finishes it will have written the IOU to ledger. Both the lender and the borrower will be able to
+     * see it when calling /spring/api/ious on their respective nodes.
+     *
+     * This end-point takes a Party name parameter as part of the path. If the serving node can't find the other party
+     * in its network map cache, it will return an HTTP bad request.
+     *
+     * The flow is invoked asynchronously. It returns a future when the flow's call() method returns.
+     */
+
+    @PostMapping(value = "create-iou", produces = arrayOf("text/plain"), headers = arrayOf("Content-Type=application/x-www-form-urlencoded"))
+    fun createIOU(request: HttpServletRequest): ResponseEntity<String> {
+        val iouValue = request.getParameter("iouValue").toInt()
+        val partyName = request.getParameter("partyName")
+        val partyX500Name = CordaX500Name.parse(partyName)
+        if (iouValue <= 0 ) {
+            return ResponseEntity.badRequest().body("Query parameter 'iouValue' must be non-negative.\n")
+        }
+        if (partyX500Name == null) {
```

Comment:
@EricMcEvoyR3 The check should be on `partyName` rather than `partyX500Name` (which cannot be null). Keep on eye on the compiler warnings :)